### PR TITLE
4000 - Utils for object manipulation

### PR DESCRIPTION
**Pull request by sorja**
Introduce two new object utils:

- mergePartial
- getDiffAsPartialObject

These will be used in upcoming PRs

### CODE REVIEW

File: src/utils/objects/getDiffAsPartialObject.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { isEmpty } from './isEmpty'
+import { setInPath } from './setInPath'
+
+const _traverseAndCompare = <T extends Record<string, unknown>>(
+  a: unknown,
+  b: unknown,
+  path: Array<string> = [],
+  keysToIgnore: Array<string> = [],
```
Comment (sorja) [inline src/utils/objects/getDiffAsPartialObject.ts]: for example, we might want to ignore the 'uuid' or 'rowId' as it is not relevant for us (in this specific case)

File: src/utils/objects/mergePartial.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * const resultObj = mergePartial(baseObj, partialObj);
+ * // resultObj: { a: 1, b: { c: 2, d: 3 } }
+ *
+ * @example
+ * const baseArr = [1, 2, { a: 3 }];
+ * const partialArr = [4, undefined, { b: 5 }];
+ * const resultArr = mergePartial(baseArr, partialArr);
+ * // resultArr: [4, 2, { a: 3, b: 5 }]
+ */
+export const mergePartial = (obj1: any, partial: any): any => {
```
Comment (sorja) [inline src/utils/objects/mergePartial.ts]: Merge the output of eg. `getDiffAsPartialObject` to a base object.

Our use case is:
base object = array of country dashboard items
partial object = region level items (eg unit is different in some deeply nested row)

Comment (minotogna) [reply src/utils/objects/mergePartial.ts]: What about lodash.merge ?

File: src/utils/objects/getDiffAsPartialObject.ts (hunk 1 above)
Comment (minotogna) [reply src/utils/objects/getDiffAsPartialObject.ts]: Can this help : https://gist.github.com/Yimiprod/7ee176597fef230d1451 ?
Shorter version maybe 🤔?

Comment (sorja) [reply src/utils/objects/getDiffAsPartialObject.ts]: Seems to have problem with arrays, would keep current version
let me know

File: src/utils/objects/mergePartial.ts (hunk 2 above)
Comment (sorja) [reply src/utils/objects/mergePartial.ts]: Replaced with lodash.merge

Seems to work as expected

File: src/utils/objects/index.ts [hunk 3]
```diff
@@ -7,24 +7,29 @@ import * as isFunction from 'lodash.isfunction'
...
 import { camelize } from './camelize'
+import { getDiffAsPartialObject } from './getDiffAsPartialObject'
 import { isEmpty } from './isEmpty'
 import { propertyOf } from './propertyOf'
 import { setInPath } from './setInPath'
 
 export const Objects = {
   camelize,
   cloneDeep,
+  getDiffAsPartialObject,
```
Comment (minotogna) [inline src/utils/objects/index.ts]: Should we call this simply getDiffs ?

**Review by minotogna — APPROVED** [review]

**Pull request merged**